src/rules/__tests__/utils.test.ts

```diff
@@ -0,0 +1,424 @@
+import { TSESLint } from '@typescript-eslint/experimental-utils';
```

Comment:
great idea to have this test 👍  My only "complaint" is that it will potentially not help us discover dead code, but probably worth it

Reply:
I think we should still be able to find deadcode, just not in these methods. I plan to split the utils up into their own files under `utils/*` at some point, which'll also help.

Reply:
iirc `jest` doesn't support any kind of "don't collect coverage from <these> files/tests" ay?

Thinking about it, we should be able to rig CI to run this test specifically in its own job, and skip it when running the coverage-collecting tests

I had a think about this, and think we should be able to do something in CI to cover that?